types/react-s-alert/index.d.ts

```diff
@@ -0,0 +1,55 @@
+// Type definitions for react-s-alert 1.3
+// Project: https://github.com/juliancwirko/react-s-alert
+// Definitions by: mitsuruog <https://github.com/mitsuruog>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+import * as React from "react";
+
+declare class SAlert extends React.Component<SAlert.SAlertProps> {}
+
+declare namespace SAlert {
+  interface SAlertConfigProps {
+    position?: string;
+    offset?: number;
+    stack?: boolean | SAlertStackProps;
+    effect?: string;
+    beep?: string | boolean | SAlertBeepProps;
+    timeout?: string | number;
+    html?: boolean;
+    onClose?: () => void;
+    onShow?: () => void;
+    customFields?: object;
+    contentTemplate?: Function;
```

Comment:
You should change this to `(...args: any[]) => any` and remove `ban-types` as a rule exception.

Alternatively, I think that you might want to use `React.ComponentType` judging from the source.

Reply:
gotcha. thanks!